### PR TITLE
Bump dependencies and fix the build on CI

### PR DESCRIPTION
**Pull request by couchbase123**
This PR includes:

- Bump dependencies based on previous dependabot PR's .
- Fix the CI Github workflows in order to run at least Java 8 version (as build.gradle is targeted to 1.8 compability mainly).
  - Which includes the tests runs embedded to the build stage with TestContainers.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -28,43 +27,31 @@ jobs:
...
+          status: ${{ job.status }}
+          notification_title: "TEST - {workflow} has {status_message} - repo_url"
+          message_format: "TEST - {emoji} *{workflow}* {status_message} in <{repo_url}|{repo}@{branch}>"
+          notify_when: "success,failure,cancelled,warnings"
         env:
-          TARGET_REPOSITORY: ${{ steps.docs_target.outputs.value }}
-          GH_TOKEN: ${{ secrets.GH_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: build/docs
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (chedim) [inline .github/workflows/gradle.yml:57]: this should be `secrets.DA_SLACK_WEBHOOK_URL`

File: build.gradle [hunk 2]
```diff
@@ -115,41 +110,22 @@ subprojects {Project subproject ->
...
-
-        // Per submodule
-        maxErrors = 1
-        maxWarnings = 10
-
-        showViolations = true
-    }
-
-    checkstyleTest.enabled = false
-
```
Comment (chedim) [inline build.gradle:148]: why does this remove checkstyle?

**Review by chedim — CHANGES_REQUESTED** [review]
Needs `SLACK_WEBHOOK_URL` replaced with `DA_SLACK_WEBHOOK_URL` and not sure why is one of the changes disables checkstyle plugin?

**Review by deniswsrosa — CHANGES_REQUESTED** [review]
- The latest version is 3.7, the app is using version 3.0. All dependencies must be the latest.
- ZERO documentation
- docs-examples must be a full micronaut app (rest endpoint storing data in the database) not just a main method.
-

File: build.gradle (hunk 2 above)
Comment (couchbase123) [reply build.gradle:148]: It's unclear the usage and local build fails with that.

The XML was introduced but sounds to not be tested.
https://github.com/couchbaselabs/couchbase-micronaut/blob/main/config/checkstyle/checkstyle.xml 

Without any change regarding the current PR, runned in clean branch from master, the issue on checkstyle also popups in the pipeline.
https://github.com/couchbaselabs/couchbase-micronaut/actions/runs/3260252338/jobs/5353681285

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (couchbase123) [reply .github/workflows/gradle.yml]: That's done, thanks.

**Comment by couchbase123** [comment]
Micronaut Dependency bumped as well to the latest version is 3.7 (was 3.0).

**Comment by couchbase123** [comment]
> * The latest version is 3.7, the app is using version 3.0. All dependencies must be the latest.
> * ZERO documentation
> * docs-examples must be a full Micronaut app (rest endpoint storing data in the database) not just a main method.
> * Bucket and Scope must be defined in the configuration. Otherwise, it will be annoying for the developer to get bucket and collection every single time. So both scope and bucket must be able be injected as well.

Bump done until 3.4.0, after there is some issue - still ongoing investigation.
Other items are not part of this PR scope. (as the goal is cover basic tests based on the current code base and add pipeline check).

**Comment by couchbase123** [comment]
Bump has been completed to 3.7.2 Micronaut version.

**Review by chedim — APPROVED** [review]

**Pull request merged**